ccpage178.py

amymhaddad opened the pull request:
There are several exercises in this file, each is separated by three spaces.

amymhaddad committed:
ccpage178.py

There are several exercises in this file, each is separated by three spaces.

amymhaddad commented on ccpage178.py:
```diff
@@ -0,0 +1,179 @@
+my_icecream = IceCreamStand('whits', 'ice cream')
+print(my_icecream.ice_cream_flavors())
+
+
```
@paulghaddad One issue that I'm running into consistently is this: when I run the program a line prints out "None" and I'm not sure why? In addition to the lines I want to print, the word "None" shows up and I can't figure out why? I notice this for each exercise, but haven't noticed it before.

paulghaddad commented on ccpage178.py:
```diff
@@ -0,0 +1,179 @@
+    
+class IceCreamStand(Restaurant):
+    """An attempt to model a specific type of restaurant, an ice cream shop"""
+    def __init__(self, name, cuisine):
```
Excellent all around! 🌵

paulghaddad commented on ccpage178.py:
```diff
@@ -0,0 +1,179 @@
+    
+    def ice_cream_flavors(self):
+        """List the ice cream flavors offered"""
+        flavors_offered = ['chocolate', 'vanilla', 'strawberry']
```
This list is already an attribute in the `__Init__` method, so you can just call that and not redefine it.

paulghaddad commented on ccpage178.py:
```diff
@@ -0,0 +1,179 @@
+    def show_admin_priviledges(self):
+        """Show the list of admin priviledges"""
+        print("These are the privledges of an Admin: ")
+        for priviledge in self.priviledges:
```
Exactly...this is what's needed in the ice cream stand class.

paulghaddad commented on ccpage178.py:
```diff
@@ -0,0 +1,179 @@
+    """An attempt to provide detail about a battery for a specific car, an electric one"""
+    def __init__(self, battery=70):
+        """Initialize the attribute"""
+        self.battery = battery
```
There's probably a better name for this, such as battery size.

paulghaddad commented on ccpage178.py:
```diff
@@ -0,0 +1,179 @@
+    """An attempt to provide detail about a battery for a specific car, an electric one"""
+    def __init__(self, battery=70):
+        """Initialize the attribute"""
+        self.battery = battery
```
I'd add an instance variable for the `range` that will store it just like the `battery_size`.

paulghaddad replied:
`amy`

```
lkasdjf
lasdflafsd
```

paulghaddad reviewed (changes requested):
Looks great, just a few things! 🐳

amymhaddad committed:
Update ccpage178.py

amymhaddad committed:
Update ccpage178.py

amymhaddad commented:
@paulghaddad This file has been updated with these changes:
-Removed trailing whitespaces; added a space after each docstring after each class definition
-Updated method beginning on line 38 to just call the attribute previously defined and removed the variable that listed all of the variables
-Updated attribute name from `self.battery` to `self.battery_size` in the method that begins on line 147
-Made `self-range` an attribute beginning on line 137.

****
Also, one issue that I'm running into consistently is this: when I run the program a line prints out "None" and I'm not sure why? In addition to the lines I want to print, the word "None" shows up and I can't figure out why? I notice this for each exercise, but haven't noticed it before.

amymhaddad committed:
Update ccpage178.py

amymhaddad committed:
Update ccpage178.py
